core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -458,16 +457,32 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     mStatistics = statistics;
     mUri = URI.create(mAlluxioHeader);
 
-    boolean masterAddIsSameAsDefault = checkMasterAddress();
-
-    if (sInitialized && masterAddIsSameAsDefault) {
+    // Create the master inquire client that we would have after merging the hadoop conf into
```

Comment:
NIT: put this new code block in a separate util method (like `checkMasterAddress`) to make this method more readable --- this  method is so critical that I want to leave it really clean.

Reply:
good idea, done